src/OpenTelemetry/Metrics/MeterProviderBuilderBase.cs

```diff
@@ -105,6 +109,22 @@ internal MeterProviderBuilder AddView(Func<Instrument, MetricStreamConfiguration
             return this;
         }
 
+        internal MeterProviderBuilder SetMetricStreamLimit(int metricStreamLimit)
```

Comment:
Just a random thought, we could combine these two methods into a single one something like:

```csharp
SetMetricLimits(int metricStreamLimit = [default], int metricPointLimit = [default])
```

Kind of reminds me of: https://docs.microsoft.com/en-us/dotnet/api/system.threading.threadpool.setmaxthreads?view=net-6.0

As far as the names, I think "min/max" might be slightly more standard than "limit"?

`maxUniqueMetricStreamCount`
`maxUniqueMetricPointCount`

Or maybe "capacity"?

`uniqueMetricStreamCapacity`
`uniqueMetricPointCapacity`

Capacity might better indicate it is pre-allocated. Common in the stream apis ex: https://docs.microsoft.com/en-us/dotnet/api/system.io.memorystream.-ctor?view=net-6.0#System_IO_MemoryStream__ctor_System_Int32_

🤷

Reply:
renaming to SetMaxMetricStreams and SetMaxMetricPointsPerMetricStream as they sound better.
Capacity might indicate the pre-allocation, but that's a implementation detail and may change.